Add FXIOS-13930 [Translations] loading + active icon

cyndichin opened the pull request:
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-13930)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

## :bulb: Description
Added loading spinner and translate active icon logic. You can test with tapping on the translate icon, see it loads and then eventually view the active icon after X amount of time. This PR does not consider the case if you tap on reload or interact with any other icon while the translation icon is in the loading state. 

- Added `templateModeforImage` and `shouldUseLoadingSpinner` to `ToolbarElemet` for now in order to show the proper icon when the state of the translation icon is updated.
- Created `TranslationsMiddleware` to handle checking whether we should show the translate icon when webpage loads, start translating page when user taps on button and revert page to normal when tapping on the active icon. Backend is not merged yet, so these are pseudo calls for now.
- Update `TranslationConfiguration` to have an inactive, loading and active state.

There are a couple of edge cases currently. Things will make more sense when tying the frontend with the backend. 
Added some TODOs to further refactor this area, but tried to keep this PR small.

cc: @issammani 

## :movie_camera: Demos

https://github.com/user-attachments/assets/64bdf629-edfc-4a86-985b-09b141f17a08

## :pencil: Checklist
- [x] I filled in the ticket numbers and a description of my work
- [x] I updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] I ensured unit tests pass and wrote tests for new code
- [ ] If working on UI, I checked and implemented accessibility (Dynamic Text and VoiceOver)
- [ ] If adding telemetry, I read the [data stewardship requirements](https://github.com/mozilla-mobile/firefox-ios/wiki/Adding-Glean-Telemetry-Events) and will request a data review
- [ ] If adding or modifying strings, I read the [guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/How-to-add-and-modify-Strings) and will request a string review from l10n
- [ ] If needed, I updated documentation and added comments to complex code

cyndichin commented:
@thatswinnie, @ih-codes Trying to break things up into smaller PRs as I can but let me know if anything is confusing and happy to hop on a quick call!

cyndichin commented:
@ih-codes @lmarceau seems like this PR is failing on danger, but I already tag ya

ih-codes commented on BrowserKit/Sources/ToolbarKit/ToolbarElement.swift:
```diff
@@ -17,6 +17,12 @@ public struct ToolbarElement: Equatable {
     let maskImageName: String?
 
+    /// Indiciated whether to use template mode or not for an image
+    let templateModeforImage: Bool
```
```suggestion
    /// Indicates whether to use template mode or not for an image
    let templateModeForImage: Bool
```

P.S. I noticed this because of Xcode's spellchecker. Worth enabling if you haven't already!!

<img width="414" height="81" alt="Image" src="https://github.com/user-attachments/assets/742e9ca6-b4f1-40ca-91a3-398f5875295e" />

ih-codes commented on firefox-ios/Client/Frontend/Translations/TranslationsMiddleware.swift:
```diff
@@ -0,0 +1,130 @@
+        }
+    }
+
+    private func handleTappingOnTranslateButtion(for action: ToolbarMiddlewareAction, and state: AppState) {
```
```suggestion
    private func handleTappingOnTranslateButton(for action: ToolbarMiddlewareAction, and state: AppState) {
```

ih-codes commented on firefox-ios/Client/Frontend/Translations/TranslationsMiddleware.swift:
```diff
@@ -0,0 +1,130 @@
+
+    // TODO: FXIOS-13844 - Start translation a page and dispatch action after completion
+    @MainActor
+    private func retrievedTranslations(for action: ToolbarMiddlewareAction) {
```
```suggestion
    private func retrieveTranslations(for action: ToolbarMiddlewareAction) {
```

ih-codes reviewed (commented):
Looks good, tested locally! 👍  I just noticed a few typos.

cyndichin commented:
@ih-codes thanks for your thorough review as always! I addressed your comments and have updated changes here - f203787639e42730d0f39757fa6089d029e7f31a
